### DISCUSSION TITLE
What is the proper way to include RAG content in an API call?

### DISCUSSION THREAD

**Discussion by intitni** [TabbyML/tabby#1584] [Q&A]
Hello, I recently brought Tabby support to Xcode through [Copilot for Xcode](https://github.com/intitni/CopilotForXcode) and its [extension](https://github.com/intitni/CustomSuggestionServiceForCopilotForXcode). It works great. 

Since I am doing RAG inside Copilot for Xcode, I am wondering what would be the correct way to include the content in the prompt. I have tried the `clipboard` field but it didn't work as expected.

Currently, I am prepending the content to the top of the `prefix`. If prepending is the way to go, what is the suggested size of `prefix`?

**Comment by wsxiaoys** [comment]
Tabby manages the RAG layer on the server side at the repository level. Therefore, you can specify the repositories of interest in the config.toml file and simply add the necessary prefixes or suffixes as they are.

P.S. Thank you for your efforts in creating Xcode support! Please let me know if you would like to be featured as a use case for Tabby's third-party editor support.

**Reply by intitni** [reply]
Yes, we would love to be featured as a use case.

I see. In Copilot for Xcode, we are also collecting snippets such as type interfaces used in the active document from the SDK, dependencies, and new code. It would be helpful if there is a way to include external RAG content.
